File: 0899-orderly-queue/0899-orderly-queue.py
class Solution:
    def orderlyQueue(self, s: str, k: int) -> str:
        if k == 0:
            return s
        if k > 1:
            return "".join(sorted(s))
        st = s
        for i in range(1, len(s)):
            st = min(st, s[i:] + s[0:i])
            
        return st 

File: 0899-orderly-queue/0899-orderly-queue.cpp
class Solution {
public:
    string orderlyQueue(string s, int k) {
        if(k == 0)
            return s;
        if(k > 1)
        {
            sort(s.begin(), s.end());
            return s;
        }
        string st = s;
        for(int i = 1; i < s.size(); i++)
        {
            string p = s.substr(i) + s.substr(0, i);
            if(st > p)
                st = p;
        }
        return st;
    }
}; 